rust-runtime/aws-smithy-types/src/event_stream.rs

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Types relevant to event stream serialization/deserialization
+
+use crate::str_bytes::StrBytes;
+use bytes::Bytes;
+
+mod value {
+    use crate::str_bytes::StrBytes;
+    use crate::DateTime;
+    use bytes::Bytes;
+
+    /// Event Stream frame header value.
+    #[non_exhaustive]
```

Comment:
I thought we only added the `#[non_exhaustive]` for errors? In future, if we introduce a new variant don't we want the consumer code to also break and have them handle the new variant?

Also, does adding a new variant to a non-exhaustive enum considered a breaking change? if yes, then I guess we will have to add `#[non_exhaustive]`.

Reply:
> does adding a new variant to a non-exhaustive enum considered a breaking change? if yes, then I guess we will have to add #[non_exhaustive].

Yes, It's for that reason.